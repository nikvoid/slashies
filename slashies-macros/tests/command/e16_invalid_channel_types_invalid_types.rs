use serenity::model::channel::PartialChannel;
use slashies_macros::Command;

/// An invalid command
#[derive(Command)]
#[name = "BadCommand"]
struct BadCommand {
    /// Field with channel types that don't exist
    #[channel_types("telegram", "telepathy")]
    bad_field: PartialChannel,

    /// Ok field
    #[channel_types("text")]
    good_field: PartialChannel,
}

fn main() {}
