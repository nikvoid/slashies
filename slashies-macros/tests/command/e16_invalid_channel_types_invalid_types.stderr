error: Invalid channel type
 --> tests/command/e16_invalid_channel_types_invalid_types.rs:9:21
  |
9 |     #[channel_types("telegram", "telepathy")]
  |                     ^^^^^^^^^^
